src/FormModel.php

```diff
@@ -214,16 +209,14 @@ public function setAttribute(string $name, $value): void
         }
     }
 
-    public function validate(): bool
+    public function validate(ValidatorInterface $validator): bool
```

Comment:
Since validator is stateless, can it be dependency passed via `__construct()`?

Reply:
Yes, but why?
Creating form model with `new` is more handy then with factory/mixing not real dependencies with real in a constructor.
```php
$form = new AuthForm($param1, $param2);
```
vs

```php
$form = new AuthForm($param1, $param2, $validator);
```

Reply:
Also I hope `validate` will be omitted and forms will validates with validator like this: `$validator->validate($form)`.

Reply:
Currently forms are obtained via type-hinting anyway:

https://github.com/yiisoft/yii-demo/blob/master/src/Contact/ContactController.php#L35

Reply:
Getting forms from DIC is wrong.

Reply:
Discussed. It's better to be as in this PR:

1. Storing form instances in container isn't a good idea anyway.
2. `$form->validate($validator->withFormatter($myFormatter))` is better when validator is available.